test/e2e/auth/audit.go

```diff
@@ -653,24 +656,33 @@ func expectAuditLines(f *framework.Framework, expected []auditEvent) {
 		expectations[event] = false
 	}
 
+	var err error
 	// Fetch the log stream.
-	stream, err := f.ClientSet.CoreV1().RESTClient().Get().AbsPath("/logs/kube-apiserver-audit.log").Stream()
-	framework.ExpectNoError(err, "could not read audit log")
-	defer stream.Close()
-
-	scanner := bufio.NewScanner(stream)
-	for scanner.Scan() {
-		line := scanner.Text()
-		event, err := parseAuditLine(line)
-		framework.ExpectNoError(err)
-
-		// If the event was expected, mark it as found.
-		if _, found := expectations[event]; found {
-			expectations[event] = true
+	for i := 0; i < retries; i++ {
+		stream, err := f.ClientSet.CoreV1().RESTClient().Get().AbsPath("/logs/kube-apiserver-audit.log").Stream()
```

Comment:
This shadows the `err` variable you've defined outside the loop.

Reply:
Maybe put this in a closure? You'd avoid shadowing and would execute the defers immediately. 

```
err = func() error {
    // execute logic
}()
if err == nil {
    break
}
```

Reply:
Thank you for suggestion. Done.